client/terraform_cli.go

```diff
@@ -85,20 +90,42 @@ func NewTerraformCLI(config *TerraformCLIConfig) (*TerraformCLI, error) {
 // Init initializes by executing the cli command `terraform init` and
 // `terraform workspace new <name>`
 func (t *TerraformCLI) Init(ctx context.Context) error {
+	var wsCreated bool
+
+	// This is special handling for when the workspace has been detected in
+	// .terraform/environment without a non-empty state. This case is common
+	// when the state for the workspace has been deleted.
+TF_INIT_AGAIN:
```

Comment:
Normally I prefer explicit loops instead of goto's but this does read pretty well. Conveys (better than a loop would) that this should work the first time, except for this one situation where we need to try again.

Reply:
👍 I rarely use goto loops but it felt even worse writing the nesting with 1 retry logic.